File: src/Bionic.Bridge.Capacitor/ClipboardBridge.cs
using System.Threading.Tasks;
using Microsoft.JSInterop;

namespace Bionic.Bridge.Capacitor {
    public static class ClipboardBridge {
        private const string ClipboardWriteFunctionName = "BionicBridge.Capacitor.Clipboard.write";
        private const string ClipboardReadFunctionName = "BionicBridge.Capacitor.Clipboard.read";

        public static Task<ClipboardReadResult> Read(ClipboardReadType clipboardReadType) =>
            JSRuntime.Current.InvokeAsync<ClipboardReadResult>(ClipboardReadFunctionName, clipboardReadType);

        public static Task Write(ClipboardWrite clipboardWrite) =>
            JSRuntime.Current.InvokeAsync<object>(ClipboardWriteFunctionName, clipboardWrite);
    }

    public enum ClipboardReadType {
        String,
        Url,
        Image
    }

    public class ClipboardReadResult {
        public string value;
    }

    public class ClipboardWrite {
        public string str;
        public string image;
        public string url;
        public string label;
    }
}

File: src/Bionic.Bridge.Capacitor.JS/src/ClipboardBridge.ts
import {ClipboardReadResult, ClipboardWrite, Plugins} from "@capacitor/core";

const {Clipboard} = Plugins;

interface IClipboard {
  read(type: number): Promise<ClipboardReadResult>;

  write(options: ClipboardWrite): Promise<void>;
}

export class ClipboardBridge implements IClipboard {
  async read(type: number): Promise<ClipboardReadResult> {
    if (!Clipboard) return Promise.reject("Clipboard not available.");
    return await Clipboard.read();
  }

  async write(data: any): Promise<void> {
    if (!Clipboard) return Promise.reject("Clipboard not available.");
    await Clipboard.write({string: data.str, image: data.image, url: data.url, label: data.label});
  }

  private static translateClipboardType(type: number): 'string' | 'url' | 'image' {
    switch (type) {
      case 1:
        return 'url';
      case 2:
        return 'image';
      default:
        return 'string';
    }
  }
}
